spyder/plugins/editor/widgets/gotoline.py

```diff
@@ -75,7 +75,7 @@ def __init__(self, editor):
     def text_has_changed(self, text):
         """Line edit's text has changed."""
         text = str(text)
-        if text:
+        if text and text != '+':
             self.lineno = int(text)
         else:
             self.lineno = None
```

Comment:
I think we could add below this line `self.lineedit.clear()` (so typing `+` is not possible/gets cleared)